test/solidity/Periphery/ReceiverStargateV2.t.sol

```diff
@@ -0,0 +1,277 @@
+        amountOutMin = amounts[1];
+
+        LibSwap.SwapData[] memory swapData = new LibSwap.SwapData[](1);
+        swapData[0] = LibSwap.SwapData({
+            callTo: address(uniswap),
+            approveTo: address(uniswap),
+            sendingAssetId: ADDRESS_USDC,
+            receivingAssetId: ADDRESS_DAI,
+            fromAmount: amountIn,
+            callData: abi.encodeWithSelector(
+                uniswap.swapExactTokensForTokens.selector,
+                amountIn,
+                amountOutMin,
+                path,
+                address(executor),
+                block.timestamp + 20 minutes
+            ),
+            requiresDeposit: true
+        });
+
+        // create a 32 bytes long offset value (stargate would attach this to our payload on src chain)
+        bytes memory offSetBytes32 = new bytes(32);
+
+        // we add 32 bytes in the beginning of the calldata since this part will be cut off by OFTComposeMsgCodec.composeMsg();
+        bytes memory payload = mergeBytes(
+            offSetBytes32,
+            abi.encode(guid, swapData, receiverAddress) // actual payload
+        );
+
+        uint64 nonce = uint64(12345645);
+        uint32 srcEid = 30102;
+        uint256 amountLD = defaultUSDCAmount;
+
+        // this is the payload that we expect to receive in our lzCompose function as parameter _message
+        callData = OFTComposeMsgCodec.encode(nonce, srcEid, amountLD, payload);
+    }
+
+    function test_contractIsSetUpCorrectly() public {
+        receiver = new ReceiverStargateV2(
+            address(this),
+            address(executor),
+            ENDPOINT_V2_MAINNET,
+            100000
+        );
+
+        assertEq(address(receiver.executor()) == address(executor), true);
+        assertEq(receiver.endpointV2() == ENDPOINT_V2_MAINNET, true);
+        assertEq(receiver.recoverGas() == RECOVER_GAS_VALUE, true);
+    }
+
+    function test_OnlyEndpointV2CanCallLzCompose() public {
+        // mock-send bridged funds to receiver contract
+        deal(ADDRESS_USDC, address(receiver), defaultUSDCAmount);
+
+        // get valid payload for calling lzCompose
+        // https://stargateprotocol.gitbook.io/stargate/v/v2-developer-docs/integrate-with-stargate/composability#receive
+        (bytes memory composeMsg, ) = _getValidLzComposeCalldata();
+
+        // call from deployer of ReceiverStargateV2
+        vm.startPrank(address(this));
+        vm.expectRevert(UnAuthorized.selector);
+
+        receiver.lzCompose{ gas: 400000 }(
+            address(0),
+            "",
+            composeMsg,
+            address(0),
+            ""
+        );
+        // call from owner of ReceiverStargateV2
+        vm.startPrank(address(this));
+        vm.expectRevert(UnAuthorized.selector);
+
+        receiver.lzCompose{ gas: 400000 }(
+            address(0),
+            "",
+            composeMsg,
+            address(0),
+            ""
+        );
```

Comment:
Lines 175-184 are identical to Lines 163-173.  Consider just updating the comment in line 163 to say "deployer/owner"?